### PR TITLE
Code restracturing to suit KenyaEMR architecture

### PR DESCRIPTION
**Pull request by Bushisky**
Code restracturing to suit KenyaEMR architecture

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -5,7 +5,7 @@
 **/.project
 **/.settings
 **/target
-*.iml
```
Comment (ningosi) [inline .gitignore]: You need this to exclude all the .iml file, just maintain *.iml

File: omod/src/main/resources/config.xml [hunk 2]
```diff
@@ -27,27 +27,31 @@
 			org.openmrs.module.uilibrary
 		</require_module>
 
-		<require_module version="1.0.2">
+		<require_module version="1.0.0">
```
Comment (ningosi) [inline omod/src/main/resources/config.xml]: You can inherit the version from the main pom file as global variables as done here https://github.com/uonafya/ehrreports/blob/master/omod/src/main/resources/config.xml#L28-L39, those variables are declared in the main pom file like this https://github.com/uonafya/ehrreports/blob/master/pom.xml#L17-L26

File: omod/src/main/resources/config.xml [hunk 3]
```diff
@@ -27,27 +27,31 @@
...
         <require_module version="2.4">
             org.openmrs.module.appframework
         </require_module>
 		
-		<require_module version="1.5.1">
+		<require_module version="1.0.0">
 			org.openmrs.module.hospitalcore
 		</require_module>
 
+		<require_module version="17.0.4">
```
Comment (ningosi) [inline omod/src/main/resources/config.xml]: KenyaEMR version we have on the test server 17.0.3

File: omod/src/main/resources/config.xml [hunk 4]
```diff
@@ -27,27 +27,31 @@
...
 		</require_module>
-
+		<!--
 		<require_module version="1.0.1">
 			org.openmrs.module.referenceapplication
 		</require_module>
-
-		<require_module version="1.0.2">
+        -->
+		<require_module version="1.0.0">
```
Comment (ningosi) [inline omod/src/main/resources/config.xml]: This version is suppossed to be the SNAPSHOT one? we don't have such a release that you are referring to

**Review by ningosi — CHANGES_REQUESTED** [review]
@Bushisky see the comments, we need to do more than this. fire up the apps that are needed in this module and put them on test server for review

File: omod/src/main/resources/config.xml (hunk 3 above)
Comment (Bushisky) [reply omod/src/main/resources/config.xml]: Actually sir its 17.0.4

File: omod/src/main/resources/config.xml (hunk 2 above)
Comment (Bushisky) [reply omod/src/main/resources/config.xml]: I am trying to inherit as requested but im getting this error from main pom. 
'dependencies.dependency.version' for org.openmrs.module:uilibrary-api:jar must be a valid version but is '${uilibraryVersion}'.

**Comment by ningosi** [comment]
Closing this, it has been updated with other references

**Pull request closed without merging**